### PR TITLE
fix(discover): Make geo columns and group_id nullable

### PR DESCRIPTION
**Pull request by Zylphrex**

### CODE REVIEW

File: snuba/datasets/entities/discover.py [hunk 1]
```diff
@@ -271,7 +271,7 @@ def attempt_map(
 
 transaction_translation_mappers = TranslationMappers(
     columns=[
-        ColumnToLiteral(None, "group_id", 0),
+        ColumnToLiteral(None, "group_id", None),
```
Comment (fpacifici) [inline snuba/datasets/entities/discover.py:274]: Why are we changing this ?

**Review by fpacifici — COMMENTED** [review]
I have nothing against this PR, though what is the context and the reason to make this change? The description is empty.

**Comment by Zylphrex** [comment]
> I have nothing against this PR, though what is the context and the reason to make this change? The description is empty.

Ah, sorry. This was still WIP so I left as a draft, with the intent to come back and write the description for it.

This PR was intended to consolidate some of the discrepancies we found when querying against the discover dataset.

1. On errors, the `geo.*` contexts were promoted and made nullable, however, they were left in the contexts array for transactions meaning the empty state was an empty string rather than Null like in errors.
2. On errors, the `GroupIdColumnProcessor` transforms the id 0 into Null https://github.com/getsentry/snuba/blob/master/snuba/datasets/storages/group_id_column_processor.py#L12-L19, but for transactions, the default was 0.

I'll be updating the description soon. Sorry about that.

**Comment by fpacifici** [comment]
ok, please see the test failure.

**Comment by Zylphrex** [comment]
Splitting this up into smaller parts. First one is the geo contexts in #1632

**Pull request closed without merging**